### PR TITLE
Custom baseurl with discovery

### PR DESCRIPTION
**Pull request by alexef**
#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [ ] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/scaffolder-actions/scaffolder-backend-module-http-request/README.md [hunk 1]
```diff
@@ -27,9 +27,9 @@ Configure the action:
 
 ```typescript
 // packages/backend/src/plugins/scaffolder.ts
-
+const discovery = UrlPatternDiscovery.compile(config.getOptionalString("backend.baseUrl") + "/{{pluginId}}")
```
Comment (punkle) [inline plugins/scaffolder-actions/scaffolder-backend-module-http-request/README.md:30]: This should already be passed in via the PluginEnvironment. There would be no need to construct it here.

See here for example in the backstage example backend.

https://github.com/backstage/backstage/blob/master/packages/backend/src/types.ts#L36

**Comment by punkle** [comment]
You would also need to change here, I think https://github.com/RoadieHQ/roadie-backstage-plugins/blob/main/packages/backend/src/plugins/scaffolder.ts#L83

File: plugins/scaffolder-actions/scaffolder-backend-module-http-request/README.md (hunk 1 above)
Comment (alexef) [reply plugins/scaffolder-actions/scaffolder-backend-module-http-request/README.md]: Updated.

File: plugins/scaffolder-actions/scaffolder-backend-module-http-request/README.md [hunk 2]
```diff
@@ -27,9 +27,8 @@ Configure the action:
 
 ```typescript
 // packages/backend/src/plugins/scaffolder.ts
-
 const actions = [
-  createHttpBackstageAction({ config }),
+  createHttpBackstageAction({discovery: env.discovery}),
```
[the remark this answers is not in the document]
Comment (alexef) [reply plugins/scaffolder-actions/scaffolder-backend-module-http-request/README.md:31]: not sure if I did it right, please have a look

[the remark this answers is not in the document]
Comment (punkle) [reply plugins/scaffolder-actions/scaffolder-backend-module-http-request/README.md:31]: I think the example can be like this:

```
createHttpBackstageAction({ discovery }),
```

**Review by punkle — APPROVED** [review]

**Pull request merged**